src/Settings/SettingsManager.cpp

```diff
@@ -88,47 +88,36 @@ void SettingsManager::loadSettings(const QString &path)
     {
         QSettings setting(path, QSettings::IniFormat);
 
-        // load most of settings
-        for (const SettingInfo &si : settingInfo)
-        {
-            if (setting.contains(si.key()) && setting.value(si.key()).isValid())
-                set(si.name, setting.value(si.key()));
-            else
-                for (const QString &old : si.old)
-                    if (setting.contains(old) && setting.value(old).isValid())
+        std::function<void(QString, const QList<SettingInfo> &)> load;
+        load = [&](QString prefix, const QList<SettingInfo> &infos) {
```

Comment:
I would not recommend making this section of code as lambda. Lambda is meant for small section of code which can be easier for user to understand. This code is way long to go into the lambda. I would instead make it a separate function.

Reply:
The reason I make these code a lambda is that the code needs to call itself. It's ok to move it into a new function, but I think that these code just belong to the loadSettings and saveSettings, and put them inline don't make the code complex. I mean, the lambda aims to provide a way to call itself. So I don't think it's necessary to move them out.

Reply:
Its very unclear to me as what do you mean by code calling itself. Do you mean recursion.

Reply:
Yes. Load the settings recursively when type is 'config'.

Reply:
Why not make a private member function?